lib/src/solver/version_solver.dart

```diff
@@ -91,13 +91,15 @@ class VersionSolver {
         [Term(PackageRange.root(_root), false)], IncompatibilityCause.root));
 
     try {
-      var next = _root.name;
-      while (next != null) {
-        _propagate(next);
-        next = await _choosePackageVersion();
-      }
+      return await _systemCache.hosted.withPrefetching(() async {
+        var next = _root.name;
+        while (next != null) {
+          _propagate(next);
+          next = await _choosePackageVersion();
+        }
 
-      return await _result();
+        return _result();
```

Comment:
`await result` for better stack trace maybe? or does that not matter?

Reply:
Hmm - not sure - I guess you are right - I'll re-add the await to be on the safe side.

Reply:
The `await` after a `return` currently impacts the behavior of a surrounding `try/catch`. I'm not entirely sure if `try/finally` is impacted. Tracking issue is https://github.com/dart-lang/sdk/issues/39204